include/envoy/network/connection.h

```diff
@@ -319,6 +319,21 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
 
 using ConnectionPtr = std::unique_ptr<Connection>;
 
+/**
+ * Connections servicing inbound connects.
+ */
+class ServerConnection : public virtual Connection {
+public:
+  /**
+   * Set the amount of time allowed for the transport socket to report that a connection is
+   * established. The provided timeout is relative to the current time. If the connection is already
+   * established, this has no effect.
```

Comment:
The comment about the connection already established doesn't seem to match the implementation.  We do rely on the Connected event being delivered after this timeout is enabled.

Reply:
Updated, PTAL.